WebDriverAgentLib/Categories/XCUIElement+FBUtilities.m

```diff
@@ -90,31 +83,37 @@ - (XCElementSnapshot *)fb_lastSnapshot
   return self.lastSnapshot;
 }
 
-- (XCElementSnapshot *)fb_snapshotWithAttributes {
-  if (![XCElementSnapshot.class respondsToSelector:@selector(snapshotAttributesForElementSnapshotKeyPaths:)]) {
+- (nullable XCElementSnapshot *)fb_snapshotWithAttributes {
+  if (![FBConfiguration shouldLoadSnapshotWithAttributes]) {
     return nil;
   }
-  [self resolve];
   
-  id defaultParameters = [[XCAXClient_iOS sharedClient] defaultParameters];
-  
-  // Names of the properties to load. There won't be lazy loading for missing properties,
-  // thus missing properties will lead to wrong results
-  NSArray<NSString *> *propertyNames = @[
-                                         @"identifier",
-                                         @"value",
-                                         @"label",
-                                         @"frame",
-                                         @"enabled",
-                                         @"elementType"
-                                         ];
+  [self resolve];
   
-  NSSet *attributes = [XCElementSnapshot snapshotAttributesForElementSnapshotKeyPaths:propertyNames];
+  static NSDictionary *defaultParameters;
+  static NSArray *axAttributes;
   
-  NSArray *axAttributes = XCAXAccessibilityAttributesForStringAttributes(attributes);
-  if (![axAttributes containsObject:FB_XCAXAIsVisibleAttribute]) {
-    axAttributes = [axAttributes arrayByAddingObject:FB_XCAXAIsVisibleAttribute];
-  }
+  static dispatch_once_t initializeAttributesAndParametersToken;
+  dispatch_once(&initializeAttributesAndParametersToken, ^{
+    defaultParameters = [[XCAXClient_iOS sharedClient] defaultParameters];
+    // Names of the properties to load. There won't be lazy loading for missing properties,
+    // thus missing properties will lead to wrong results
+    NSArray<NSString *> *propertyNames = @[
+                      @"identifier",
+                      @"value",
+                      @"label",
+                      @"frame",
+                      @"enabled",
+                      @"elementType"
+                      ];
+    
+    NSSet *attributes = [XCElementSnapshot snapshotAttributesForElementSnapshotKeyPaths:propertyNames];
+    
+    axAttributes = XCAXAccessibilityAttributesForStringAttributes(attributes);
```

Comment:
is this attribute still needed?

Reply:
Yes, we still check if it is in the snapshot.

> however if this is really async and _XCT_snapshotForElement times out properly then it should not be a problem
If this happens the visibility won't be in the snapshot initially and we continue with the other approach